lib/puppet/provider/cisco_interface/cisco.rb

```diff
@@ -201,6 +216,74 @@ def ipv4_addr_mask_set
     end
   end
 
+  def stp_mst_cost
+    return @property_hash[:stp_mst_cost] if @resource[:stp_mst_cost].nil?
+    if @resource[:stp_mst_cost][0] == :default &&
+       @property_hash[:stp_mst_cost] == @interface.default_stp_mst_cost
+      return [:default]
+    else
+      @property_hash[:stp_mst_cost]
+    end
+  end
+
+  def stp_mst_cost=(should_list)
+    should_list = @interface.default_stp_mst_cost if should_list[0] == :default
+    # check for overlapping arrays in should_list
+    PuppetX::Cisco::Utils.fail_array_overlap(should_list)
```

Comment:
It doesn't work in the type (we discussed this in one of the scrums before). The problem is only one array is coming to the type method and here I need to see all the arrays within the bigger array to see if there are any overlaps. It only worked here.

The error I am throwing is:
"overlapping arrays not allowed"
Check https://github.com/saichint/cisco-network-puppet-module/blob/feature/stp_global/lib/puppet_x/cisco/cmnutils.rb#L60

If this error is too bland, please suggest a better one